File: app/src/test/java/org/schulcloud/mobile/data/model/SubmissionTest.java
package org.schulcloud.mobile.data.model;

import org.junit.Before;
import org.junit.Test;

import io.realm.RealmList;

import static junit.framework.Assert.assertEquals;

public class SubmissionTest {
    private static final String ID = "ID";
    private static final String SCHOOLDID = "schoolId";
    private static final String STUDENTID = "studentId";
    private static final String COMMENT = "comment";
    private static final RealmList<Comment> COMMENTS = new RealmList<>();
    private static final Integer GRADE = 99;
    private static final String HOMEWORKID = "homeworkId";
    private static final String GRADECOMMENT = "cool";
    private static final String CREATEDAT = "invisible";

    private Submission submission;

    @Before
    public void setUp() throws Exception {
        submission = createNewSubmission();
    }

    @Test
    public void testGetProperties() {
        assertEquals(submission._id, ID);
        assertEquals(submission.schoolId, SCHOOLDID);
        assertEquals(submission.studentId, STUDENTID);
        assertEquals(submission.comment, COMMENT);
        assertEquals(submission.gradeComment, GRADECOMMENT);
        assertEquals(submission.createdAt, CREATEDAT);
        assertEquals(submission.comments, COMMENTS);
        assertEquals(submission.homeworkId, HOMEWORKID);
        assertEquals(submission.grade, GRADE);
    }

    public static Submission createNewSubmission() {
        Submission s = new Submission();
        s._id = ID;
        s.schoolId = SCHOOLDID;
        s.studentId = STUDENTID;
        s.comment = COMMENT;
        s.comments = COMMENTS;
        s.gradeComment = GRADECOMMENT;
        s.grade = GRADE;
        s.homeworkId = HOMEWORKID;
        s.createdAt = CREATEDAT;

        return s;
    }
}

File: app/src/test/java/org/schulcloud/mobile/models/homework/submission/SubmissionTest.kt
package org.schulcloud.mobile.models.homework.submission

import org.junit.Assert.*
import io.realm.RealmList
import org.junit.Before
import org.junit.Test


class SubmissionTest {
    private companion object {
        const val ID = "id"
        const val HOMEWORKID = "homeworkId"
        const val STUDENTID = "studentId"
        const val COMMENT = "comment"
        const val CREATEDAT = "createdAt"
        const val GRADE = 15
        const val GRADECOMMENT = "gradeComment"
        val COMMENTS = RealmList<Comment>()
    }

    private val newSubmission: Submission
        get() = Submission().apply {
            id = ID
            homeworkId = HOMEWORKID
            studentId = STUDENTID
            comment = COMMENT
            createdAt = CREATEDAT
            grade = GRADE
            gradeComment = GRADECOMMENT
            comments = COMMENTS
        }
    private lateinit var submission: Submission

    @Before
    fun setUp() {
        submission = newSubmission
    }

    @Test
    fun testGetProperties() {
        assertEquals(ID, submission.id)
        assertEquals(HOMEWORKID, submission.homeworkId)
        assertEquals(STUDENTID, submission.studentId)
        assertEquals(COMMENT, submission.comment)
        assertEquals(CREATEDAT, submission.createdAt)
        assertEquals(GRADE, submission.grade)
        assertEquals(GRADECOMMENT, submission.gradeComment)
        assertEquals(COMMENTS, submission.comments)
    }
}
